libraries/AC_AutoTune/AC_AutoTune_Heli.cpp

```diff
@@ -706,6 +706,31 @@ void AC_AutoTune_Heli::save_tuning_gains()
     reset();
 }
 
+// report final gains for a given axis to GCS
+void AC_AutoTune_Heli::report_final_gains(AxisType test_axis) const
+{
+    switch (test_axis) {
+        case ROLL:
+            report_axis_gains("Roll", tune_roll_rp, tune_roll_rff*AUTOTUNE_FFI_RATIO_FINAL, tune_roll_rd, tune_roll_rff, tune_roll_sp, tune_roll_accel);
+            break;
+        case PITCH:
+            report_axis_gains("Pitch", tune_pitch_rp, tune_pitch_rff*AUTOTUNE_FFI_RATIO_FINAL, tune_pitch_rd, tune_pitch_rff, tune_pitch_sp, tune_pitch_accel);
+            break;
+        case YAW:
+            report_axis_gains("Yaw", tune_yaw_rp, tune_yaw_rp*AUTOTUNE_YAW_PI_RATIO_FINAL, tune_yaw_rd, tune_yaw_rff, tune_yaw_sp, tune_yaw_accel);
+            break;
+    }
+}
+
+// report gain formating helper
+void AC_AutoTune_Heli::report_axis_gains(const char* axis_string, float rate_P, float rate_I, float rate_D, float rate_ff, float angle_P, float max_rate) const
+{
+    gcs().send_text(MAV_SEVERITY_NOTICE,"AutoTune: %s complete", axis_string);
+    gcs().send_text(MAV_SEVERITY_NOTICE,"AutoTune: %s Rate: P:%0.4f, I:%0.4f, D:%0.5f, FF:%0.4f",axis_string,rate_P,rate_I,rate_D,rate_ff);
+    gcs().send_text(MAV_SEVERITY_NOTICE,"AutoTune: %s Angle P:%0.2f, Max Rate:%0.0f",axis_string,angle_P,max_rate);
```

Comment:
I think you can leave off max rate in this print statement.  I am not sure where you got this.  The other thing that could be tuned but is not in heli is the max accel parameter.  I don't change the max accel param but it does provide the user with the max accel attainable.  
So delete the max rate out of this message.  otherwise it looks good.

Reply:
@bnsgeyer You set the max accel here for roll for example: https://github.com/ArduPilot/ardupilot/blob/b44442032996917f45b44c164d5ead518bf6c470/libraries/AC_AutoTune/AC_AutoTune_Heli.cpp#L637

But your right, I have miss-named it here, should be max accel not max rate.

Reply:
I do set it but it really isn't tuned.  It takes on the original value and then sets it at the end.  It is probably something that will be tuned in the future but not currently

Reply:
@bnsgeyer The goal is to print exactly the values that would be saved if the tune was finished completed.

It maybe that we need not save the accel values, but currently we do. A future PR might be to stop saving the value and remove it from the send text. 

There is some tidying up to do in this area to make sure both heli and copter autotune put the ancillary params back to how they found them, currently they do not in all cases. For example if you have body frame feedfoward disabled and don't save the autotune currently it will leave your accel max at 0.